src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -584,39 +584,37 @@ using LangMatches =
 
 // STRING WITH LANGUAGE TAG
 [[maybe_unused]] inline auto strLangTag =
-    [](std::optional<std::string> input,
+    [](std::optional<ad_utility::triple_component::Literal> literal,
        std::optional<std::string> langTag) -> IdOrLiteralOrIri {
-  if (!input.has_value() || !langTag.has_value()) {
+  if (!literal.has_value() || !langTag.has_value() ||
+      literal.value().hasLanguageTag() || literal.value().hasDatatype()) {
```

Comment:
These cases are not completely tested, 
please add unit tests that also consider all possible cases in which this `makeUndefined` can be triggered to make the coverage happy.
Same for the other function below.